compiler/rustc_middle/src/query/plumbing.rs

```diff
@@ -340,10 +340,10 @@ macro_rules! define_callbacks {
                     <$($K)* as keys::Key>::CacheSelector as CacheSelector<'tcx, Erase<$V>>
                 >::Cache;
 
-                // Ensure that keys grow no larger than 64 bytes
+                // Ensure that keys grow no larger than 72 bytes
```

Comment:
Is this not a hard limit? Why was this added in the first place?

Reply:
probably perf, but I'm trying to resolve it anyway before going out of draft

Reply:
So I'm a bit stuck on CanonicalTypeOpAscribeUserTypeGoal. It's 8 bytes too large now, but there's also nothing that stands out as "this should be interned". Just moving something onto a TyCtxt arena doesn't really seem right to me, that's just avoiding the above size check without adhering to its spirit.